Content.Shared/_Goobstation/StationEvents/Metric/ChaosMetrics.cs

```diff
@@ -0,0 +1,190 @@
+using System.Linq;
+using Content.Shared.FixedPoint;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Generic;
+using Robust.Shared.Utility;
+
+namespace Content.Shared._Goobstation.StationEvents.Metric;
+
+public enum ChaosMetric
+{
+    Hunger,
+    Thirst,
+    Charge,
+
+    Anomaly,
+
+    Friend,   // Negative score for how many friendlies on station
+    Hostile,  // Increases with hostiles
+
+    Death,
+    Medical,
+
+    Security,
+    Power,
+    Atmos,
+    Mess,
+
+    // Metrics calculated from above by Game Director:
+    Combat,   // Friend + Hostile - <0 if crew is strong. 0 if balanced (fighting). >0 indicates crew is losing.
+}
+
+/// <summary>
+///   Raised to request metrics to calculate and sum their statistics
+/// </summary>
+[ByRefEvent]
+public record struct CalculateChaosEvent(ChaosMetrics Metrics);
+
+/// <summary>
+///   This class represents a collection of chaos
+/// </summary>
+/// <remarks>
+///   This is similar to the DamageSpecifier class.
+/// </remarks>
+[DataDefinition]
+public sealed partial class ChaosMetrics : IEquatable<ChaosMetrics>
+{
+    /// <summary>
+    ///   Main chaos dictionary. Most ChaosMetrics functions exist to somehow modifying this.
+    /// </summary>
+    [IncludeDataField(customTypeSerializer:typeof(DictionarySerializer<ChaosMetric, FixedPoint2>)), ViewVariables(VVAccess.ReadWrite)]
+    public Dictionary<ChaosMetric, FixedPoint2> ChaosDict { get; set; } = new();
+
+    /// <summary>
+    ///   Whether this chaos specifier has any entries.
+    /// </summary>
+    public bool Empty => ChaosDict.Count == 0;
+
+    /// <summary>
+    ///   True if any of our values are greater than other, i.e worse.
+    /// </summary>
+    public bool AnyWorseThan(ChaosMetrics other) => ChaosDict.Any(kvp =>
+        other.ChaosDict.TryGetValue(kvp.Key, out var otherV)
+        && kvp.Value > otherV);
+
+    /// <summary>
+    ///   True if all of our values are less than other, i.e better.
+    /// </summary>
+    public bool AllBetterThan(ChaosMetrics other) => ChaosDict.All(kvp =>
+        !other.ChaosDict.TryGetValue(kvp.Key, out var otherV)
+        || kvp.Value < otherV);
```

Comment:
BLock body for slightly better legibility.
Can also make one function just be `=> !OtherOne(...)`
```suggestion
    public bool AnyWorseThan(ChaosMetrics other)
    {
        return ChaosDict.Any(kvp =>
            other.ChaosDict.TryGetValue(kvp.Key, out var otherV) && kvp.Value > otherV);
    }

    /// <summary>
    ///   True if all of our values are less than other, i.e better.
    /// </summary>
    public bool AllBetterThan(ChaosMetrics other)
    {
        return ChaosDict.All(kvp =>
            !other.ChaosDict.TryGetValue(kvp.Key, out var otherV) || kvp.Value < otherV);
    }
```